src/mobster/cmd/generate/product.py

```diff
@@ -2,22 +2,215 @@
+        output_path: Path = self.cli_args.output.joinpath(fname)
+        LOGGER.info("Saving SBOM to %s.", output_path)
+        write_file(document=self.document, file_name=str(output_path), validate=True)
+        print(output_path.absolute())
+
+
+def create_sbom(release_notes: ReleaseNotes, snapshot: Snapshot) -> Document:
+    """
+    Create an SPDX document based on release notes and a snapshot.
+    """
+    doc_elem_id = "SPDXRef-DOCUMENT"
+    product_elem_id = "SPDXRef-product"
+
+    creation_info = spdx.get_creation_info(
+        f"{release_notes.product_name} {release_notes.product_version}"
+    )
+
+    product_package = create_product_package(product_elem_id, release_notes)
+    product_relationship = create_product_relationship(doc_elem_id, product_elem_id)
+
+    component_packages = get_component_packages(snapshot.components)
+    component_relationships = get_component_relationships(
+        product_elem_id, component_packages
+    )
+
+    return Document(
+        creation_info=creation_info,
+        packages=[product_package, *component_packages],
+        relationships=[product_relationship, *component_relationships],
+    )
+
+
+def create_product_package(
+    product_elem_id: str, release_notes: ReleaseNotes
+) -> Package:
+    """Create SPDX package corresponding to the product."""
+    if isinstance(release_notes.cpe, str):
+        cpes = [release_notes.cpe]
+    else:
+        cpes = release_notes.cpe
+
+    refs = [
+        ExternalPackageRef(
+            category=ExternalPackageRefCategory.SECURITY,
+            reference_type="cpe22Type",
+            locator=cpe,
+        )
+        for cpe in cpes
+    ]
+
+    return Package(
+        spdx_id=product_elem_id,
+        name=release_notes.product_name,
+        version=release_notes.product_version,
+        download_location=SpdxNoAssertion(),
+        supplier=Actor(ActorType.ORGANIZATION, "Red Hat"),
+        license_declared=SpdxNoAssertion(),
+        files_analyzed=False,
+        external_references=refs,
+    )
+
+
+def create_product_relationship(doc_elem_id: str, product_elem_id: str) -> Relationship:
+    """Create SPDX relationship corresponding to the product SPDX package."""
+    return Relationship(
+        spdx_element_id=doc_elem_id,
+        relationship_type=RelationshipType.DESCRIBES,
+        related_spdx_element_id=product_elem_id,
+    )
+
+
+def get_component_packages(components: List[Component]) -> List[Package]:
+    """
+    Get a list of SPDX packages - one per each component.
+
+    Each component can have multiple external references - purls.
+    """
+    packages = []
+    for component in components:
+        checksum = component.image.digest.split(":", 1)[1]
+
+        purls = [
+            construct_purl(
+                component.release_repository, component.image.digest, tag=tag
+            )
+            for tag in component.tags
+        ]
+
+        packages.append(
+            Package(
```

Comment:
Could you please move this to the spdx module so we can have all package definitions in a single place? Function like `get_package_from_snapshot_component`  or similar should be a good fit.